TESTING.md

```diff
@@ -169,8 +169,99 @@ cargo make \
   test
 ```
 
-(`-r` tells testsys to also show the status of resources like the cluster and instances in addition to tests):
-
 ```shell
 cargo make watch-test
 ```
+
+## Migration Testing
+
+Migration testing is used to ensure Bottlerocket can update from one version to a new version and back. This involves launching Bottlerocket instances, upgrading them, and downgrading them.
+
+Migration testing launches instances of a starting Bottlerocket version, or a provided initial AMI and migrates instances to the target version.
+In order to accomplish this a few artifacts need to be created:
+* A publicly accessible TUF repository
+* A previous release of Bottlerocket signed with available keys
+* The AMI ID for the previous release
+* Image artifacts and local TUF repos of said artifacts for current changes
```

Comment:
The current implementation addresses my concern here, thanks!

By automating the lookup of the previous version and the AMI, it's more likely that the developer will follow the expected steps rather than going "off script" to force the use of AMIs built some other way.